nat/traversal/pinger.go

```diff
@@ -33,69 +36,92 @@ const prefix = "[NATPinger] "
 const pingInterval = 200
 const pingTimeout = 10000
 
-// ConfigParser is able to parse a config from given raw json
-type ConfigParser interface {
-	Parse(config json.RawMessage) (ip string, port int, err error)
-}
-
 // Pinger represents NAT pinger structure
 type Pinger struct {
-	localPort      int
-	pingTarget     chan json.RawMessage
-	pingReceived   chan struct{}
+	pingTarget     chan *Params
 	pingCancelled  chan struct{}
 	natEventWaiter NatEventWaiter
 	configParser   ConfigParser
-	stop           chan struct{}
 	once           sync.Once
+	natProxy       natProxy
+	portPool       portSupplier
+	consumerPort   int
 }
 
 // NatEventWaiter is responsible for waiting for nat events
 type NatEventWaiter interface {
 	WaitForEvent() Event
 }
 
+// ConfigParser is able to parse a config from given raw json
+type ConfigParser interface {
+	Parse(config json.RawMessage) (ip string, port int, serviceType services.ServiceType, err error)
+}
+
+type portSupplier interface {
```

Comment:
We gave this interface in multiple places. Can't we define it once somewhere? 
There are many examples of such definitions in a standard library: https://golang.org/pkg/io/#Reader